feat: update docs for edgex 3.0 update (#152)
intel-retail/automated-vending: docs_src/configuration.md

# Configuration

This page lists all of the relevant configuration parameters for each service in the Automated Vending reference implementation.

!!!info

>     Note that this document likely does not cover EdgeX-specific configuration parameters. Application and device service SDK documentation can be found in the [EdgeX Foundry GitHub repositories](https://github.com/edgexfoundry) or in the [official EdgeX documentation here](https://docs.edgexfoundry.org/2.2/).

Reviewer: ```suggestion
    Note that this document likely does not cover EdgeX-specific configuration parameters. Application and device service SDK documentation can be found in the [EdgeX Foundry GitHub repositories](https://github.com/edgexfoundry) or in the [official EdgeX documentation here](https://docs.edgexfoundry.org/3.0/).
```

## Environment overrides

> The simplest way to change one of the configuration values described below is via the use of environment variable overrides in the docker compose file. The value of each configuration item in a service's configuration can be overridden with an environment variable specific to that item. The name of the environment variable is the path to the item in the configuration tree with underscores separating the nodes. The character case of each node in the environment variable name must match that found in the service's configuration. These overrides are placed in the target service's environment section of the compose file. Here is an example:

Reviewer 2: You removed the `values described below` need so reference/example here.

configuration.yaml
```yaml
Driver:
  VID: "65535" # 0xFFFF
  PID: "53"    # 0x0035
```

compose file
```yaml
DRIVER_VID: "256" 
DRIVER_PID: "26"  
```

Reviewer 2: Right, They are not in the current as this is just an example of how to change configuration via env overrides.

> The simplest way to change one of the configuration values described below is via the use of environment variable overrides in the docker compose file. The value of each configuration item in a service's configuration can be overridden with an environment variable specific to that item. The name of the environment variable is the path to the item in the configuration tree with underscores separating the nodes. The character case of each node in the environment variable name must match that found in the service's configuration. These overrides are placed in the target service's environment section of the compose file. Here is an example:

Reviewer 2: Nothing actually described below...
```suggestion
The simplest way to change one of the service's configuration values is via the use of environment variable overrides in the docker compose file. The value of each configuration item in a service's configuration can be overridden with an environment variable specific to that item. The name of the environment variable is the path to the item in the configuration tree with underscores separating the nodes. The  environment variable name is upper case version of the setting that found in the service's configuration. These overrides are placed in the target service's environment section of the compose file. Here is an example:
```

      - /dev/input:/dev/input
    environment:
      DRIVERCONFIG_SIMULATEDEVICE: "false"

>       DRIVERCONFIG__DEVICESEARCHPATH: "/dev/input/event*"

Reviewer 2: Only one underscore

```suggestion
      DRIVERCONFIG_DEVICESEARCHPATH: "/dev/input/event*"
```

## EdgeX MQTT device service

> This reference implementation uses the [MQTT Device Service](https://github.com/edgexfoundry/device-mqtt-go) from EdgeX with custom device profiles. These device profiles YAML files are located [here](https://github.com/intel-retail/automated-vending/blob/Edgex-3.0/res/device-mqtt/profiles/inference-mqtt-device-profile.yml) and are volume mounted into the device service's running Docker container.

Reviewer 2: This one exists on main, so link to main branch

> The following items can be configured via `device-mqtt.environment` section of the service's [docker-compose.edgex.yml](https://github.com/intel-retail/automated-vending/tree/main/docker-compose.edgex.yml) file.

Reviewer 2: ```suggestion
The following `MQTTBrokerInfo` configuration items can be configured via `device-mqtt.environment` section of the service's [docker-compose.edgex.yml](https://github.com/intel-retail/automated-vending/tree/main/docker-compose.edgex.yml) file.
```

> The following items can be configured via `device-mqtt.environment` section of the service's [docker-compose.edgex.yml](https://github.com/intel-retail/automated-vending/blob/Edgex-3.0/docker-compose.edgex.yml) file.

Reviewer 2: compose file exist on main, so like to main branch

> `device-mqtt.environment`

Reviewer 2: ```suggestion
`MQTTBrokerInfo via device-mqtt.environment`
```

`MQTTBrokerInfo`
`device-mqtt.environment`

- `Schema` - Data schema type, aka protocol
- `Host` - Host name of the response MQTT Broker
- `Port` - Port number of the response MQTT Broker
- `Qos` - Quality of service agreement between sender and receiver
- `KeepAlive` - Keep alive duration for the response MQTT Broker
- `ClientId` - Client ID for the response MQTT Broker

> - `ResponseTopic` - Subscribe topic for the response MQTT Broker

Reviewer 2: The purpose of keeping these is an example if user wants to also change them. Please restore them in env override format.
